tests/phpunit/tests/admin/wpUpgrader.php

```diff
@@ -0,0 +1,1430 @@
+<?php
+/**
+ * Tests the `WP_Upgrader` class.
+ *
+ * @package WordPress
+ *
+ * @since 6.1.0
+ */
+
+/**
+ * Tests_Admin_WpUpgrader class.
+ *
+ * @group admin
+ * @group upgrade
+ *
+ * @covers WP_Upgrader
+ *
+ * @since 6.1.0
+ */
+class Tests_Admin_WpUpgrader extends WP_UnitTestCase {
+
+	/**
+	 * An instance of the WP_Upgrader class being tested.
+	 *
+	 * @var WP_Upgrader
+	 */
+	private $instance;
+
+	/**
+	 * @var WP_Upgrader_Skin&\PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $upgrader_skin_mock;
+
+	/**
+	 * Filesystem mock.
+	 *
+	 * @var \PHPUnit\Framework\MockObject\MockObject&WP_Filesystem_Base
```

Comment:
```suggestion
	 * @var WP_Filesystem_Base&\PHPUnit\Framework\MockObject\MockObject
```
(order was inconsistent with the property above it)

Also - for both: why is the PHPUnit class a FQN with a leading backslash, while the WP native class is an FQN without the leading backslash ?

Reply:
These were taken from the original PR and, admittedly, I didn't review the types.

To confirm, would you recommend adding the leading backslash to the WP native class, or removing it from the PHPUnit class?

Reply:
Considering it is WP Core and no namespaces are used in Core, I'd say "remove". In all other cases, I'd probably say "add"... 🤷🏻‍♀️